HW03-Part2.Rmd

```diff
@@ -0,0 +1,209 @@
+---
+title: "Homework 03: Create Visualization"
+output: github_document
+---
+
+```{r setup, include=FALSE}
```

Comment:
Cleaned up appearance and more simply coded (see my comments on Canvas once I post grades)

``` r
library(tidyverse)

# import data
dc <- read_csv("data/dc-wikia-data.csv")
#> Parsed with column specification:
#> cols(
#>   brand = col_character(),
#>   page_id = col_double(),
#>   name = col_character(),
#>   urlslug = col_character(),
#>   ID = col_character(),
#>   ALIGN = col_character(),
#>   EYE = col_character(),
#>   HAIR = col_character(),
#>   SEX = col_character(),
#>   GSM = col_character(),
#>   ALIVE = col_character(),
#>   APPEARANCES = col_double(),
#>   `FIRST APPEARANCE` = col_character(),
#>   YEAR = col_double()
#> )
marvel <- read_csv("data/marvel-wikia-data.csv")
#> Parsed with column specification:
#> cols(
#>   brand = col_character(),
#>   page_id = col_double(),
#>   name = col_character(),
#>   urlslug = col_character(),
#>   ID = col_character(),
#>   ALIGN = col_character(),
#>   EYE = col_character(),
#>   HAIR = col_character(),
#>   SEX = col_character(),
#>   GSM = col_character(),
#>   ALIVE = col_character(),
#>   APPEARANCES = col_double(),
#>   `FIRST APPEARANCE` = col_character(),
#>   YEAR = col_double()
#> )
comics <- bind_rows(dc, marvel)

# generate faceted plot
comics %>%
  # minimize data frame to only essential rows/columns
  group_by(brand, SEX, YEAR, ID) %>%
  select() %>%
  drop_na() %>%
  filter(SEX == "Male Characters" | SEX == "Female Characters") %>%
  # calculate proportion of secret characters by brand, sex, and year
  group_by(brand, SEX, YEAR) %>%
  summarize(pct_secret = mean(ID == "Secret Identity")) %>%
  # fill in years with no observations as NA to draw gaps in graph
  ungroup() %>%
  complete(brand, SEX, YEAR) %>%
  # generate the plot
  ggplot(mapping = aes(x = YEAR, y = pct_secret)) +
  geom_line() +
  scale_y_continuous(labels = scales::percent) +
  labs(title = "The change in comic book character's identity secrecy over time",
       x = "Year",
       y = "Percent of characters with secret identities") +
  facet_grid(brand ~ SEX, scales = "free") +
  theme_minimal()
#> Adding missing grouping variables: `brand`, `SEX`, `YEAR`, `ID`
#> Warning: Removed 1 rows containing missing values (geom_path).
```

![](https://i.imgur.com/QzoTLqN.png)

<sup>Created on 2019-05-31 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.5.3 (2019-03-11)
#>  os       macOS Mojave 10.14.5        
#>  system   x86_64, darwin15.6.0        
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/Los_Angeles         
#>  date     2019-05-31                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package     * version date       lib source        
#>  assertthat    0.2.1   2019-03-21 [2] CRAN (R 3.5.3)
#>  backports     1.1.4   2019-04-10 [2] CRAN (R 3.5.2)
#>  broom         0.5.2   2019-04-07 [2] CRAN (R 3.5.2)
#>  callr         3.2.0   2019-03-15 [2] CRAN (R 3.5.2)
#>  cellranger    1.1.0   2016-07-27 [2] CRAN (R 3.5.0)
#>  cli           1.1.0   2019-03-19 [1] CRAN (R 3.5.2)
#>  colorspace    1.4-1   2019-03-18 [2] CRAN (R 3.5.2)
#>  crayon        1.3.4   2017-09-16 [2] CRAN (R 3.5.0)
#>  curl          3.3     2019-01-10 [2] CRAN (R 3.5.2)
#>  desc          1.2.0   2018-05-01 [2] CRAN (R 3.5.0)
#>  devtools      2.0.2   2019-04-08 [1] CRAN (R 3.5.2)
#>  digest        0.6.19  2019-05-20 [1] CRAN (R 3.5.2)
#>  dplyr       * 0.8.1   2019-05-14 [1] CRAN (R 3.5.2)
#>  evaluate      0.13    2019-02-12 [2] CRAN (R 3.5.2)
#>  forcats     * 0.4.0   2019-02-17 [2] CRAN (R 3.5.2)
#>  fs            1.3.1   2019-05-06 [1] CRAN (R 3.5.2)
#>  generics      0.0.2   2018-11-29 [1] CRAN (R 3.5.0)
#>  ggplot2     * 3.1.1   2019-04-07 [1] CRAN (R 3.5.2)
#>  glue          1.3.1   2019-03-12 [2] CRAN (R 3.5.2)
#>  gtable        0.3.0   2019-03-25 [2] CRAN (R 3.5.2)
#>  haven         2.1.0   2019-02-19 [2] CRAN (R 3.5.2)
#>  highr         0.8     2019-03-20 [2] CRAN (R 3.5.3)
#>  hms           0.4.2   2018-03-10 [2] CRAN (R 3.5.0)
#>  htmltools     0.3.6   2017-04-28 [1] CRAN (R 3.5.0)
#>  httr          1.4.0   2018-12-11 [2] CRAN (R 3.5.0)
#>  jsonlite      1.6     2018-12-07 [2] CRAN (R 3.5.0)
#>  knitr         1.22    2019-03-08 [2] CRAN (R 3.5.2)
#>  labeling      0.3     2014-08-23 [2] CRAN (R 3.5.0)
#>  lattice       0.20-38 2018-11-04 [2] CRAN (R 3.5.3)
#>  lazyeval      0.2.2   2019-03-15 [2] CRAN (R 3.5.2)
#>  lubridate     1.7.4   2018-04-11 [2] CRAN (R 3.5.0)
#>  magrittr      1.5     2014-11-22 [2] CRAN (R 3.5.0)
#>  memoise       1.1.0   2017-04-21 [2] CRAN (R 3.5.0)
#>  mime          0.6     2018-10-05 [1] CRAN (R 3.5.0)
#>  modelr        0.1.4   2019-02-18 [2] CRAN (R 3.5.2)
#>  munsell       0.5.0   2018-06-12 [2] CRAN (R 3.5.0)
#>  nlme          3.1-140 2019-05-12 [2] CRAN (R 3.5.2)
#>  pillar        1.4.0   2019-05-11 [2] CRAN (R 3.5.2)
#>  pkgbuild      1.0.3   2019-03-20 [1] CRAN (R 3.5.3)
#>  pkgconfig     2.0.2   2018-08-16 [2] CRAN (R 3.5.1)
#>  pkgload       1.0.2   2018-10-29 [1] CRAN (R 3.5.0)
#>  plyr          1.8.4   2016-06-08 [2] CRAN (R 3.5.0)
#>  prettyunits   1.0.2   2015-07-13 [2] CRAN (R 3.5.0)
#>  processx      3.3.1   2019-05-08 [1] CRAN (R 3.5.2)
#>  ps            1.3.0   2018-12-21 [2] CRAN (R 3.5.0)
#>  purrr       * 0.3.2   2019-03-15 [2] CRAN (R 3.5.2)
#>  R6            2.4.0   2019-02-14 [1] CRAN (R 3.5.2)
#>  Rcpp          1.0.1   2019-03-17 [1] CRAN (R 3.5.2)
#>  readr       * 1.3.1   2018-12-21 [2] CRAN (R 3.5.0)
#>  readxl        1.3.1   2019-03-13 [2] CRAN (R 3.5.2)
#>  remotes       2.0.4   2019-04-10 [1] CRAN (R 3.5.2)
#>  reshape2      1.4.3   2017-12-11 [2] CRAN (R 3.5.0)
#>  rlang         0.3.4   2019-04-07 [1] CRAN (R 3.5.2)
#>  rmarkdown     1.12    2019-03-14 [1] CRAN (R 3.5.2)
#>  rprojroot     1.3-2   2018-01-03 [2] CRAN (R 3.5.0)
#>  rvest         0.3.4   2019-05-15 [2] CRAN (R 3.5.2)
#>  scales        1.0.0   2018-08-09 [1] CRAN (R 3.5.0)
#>  sessioninfo   1.1.1   2018-11-05 [1] CRAN (R 3.5.0)
#>  stringi       1.4.3   2019-03-12 [1] CRAN (R 3.5.2)
#>  stringr     * 1.4.0   2019-02-10 [1] CRAN (R 3.5.2)
#>  testthat      2.1.1   2019-04-23 [2] CRAN (R 3.5.2)
#>  tibble      * 2.1.1   2019-03-16 [2] CRAN (R 3.5.2)
#>  tidyr       * 0.8.3   2019-03-01 [1] CRAN (R 3.5.2)
#>  tidyselect    0.2.5   2018-10-11 [1] CRAN (R 3.5.0)
#>  tidyverse   * 1.2.1   2017-11-14 [2] CRAN (R 3.5.0)
#>  usethis       1.5.0   2019-04-07 [1] CRAN (R 3.5.2)
#>  withr         2.1.2   2018-03-15 [2] CRAN (R 3.5.0)
#>  xfun          0.7     2019-05-14 [1] CRAN (R 3.5.2)
#>  xml2          1.2.0   2018-01-24 [2] CRAN (R 3.5.0)
#>  yaml          2.2.0   2018-07-25 [2] CRAN (R 3.5.0)
#> 
#> [1] /Users/soltoffbc/Library/R/3.5/library
#> [2] /Library/Frameworks/R.framework/Versions/3.5/Resources/library
```

</details>